src/client.js

```diff
@@ -1764,6 +1768,44 @@ class Client {
         return result;
     }
 
+    /**
+     * This is the exposed/public method to request context data for a specific report.
+     * @param {*} callContext: {reportName: string, context: string, selection: string, selectionCount: number, formData: any}
+     * @param {string} representation the expected representation ('xml', 'BadgerFish', or 'SimpleJson'). If not set, will use the current representation
+     *
+     * @returns {Campaign.ReportContext} an object containing the context data for a specific report
+     */
+    async getReport(callContext, representation) {
+        try {
+            if(callContext.formData?.ctx) {
+                var xmlCtx = this._fromRepresentation('ctx', callContext.formData.ctx);
+                var formData = {...callContext.formData, ctx: DomUtil.toXMLString(xmlCtx)};
+            }
+            const request = {
+                url: `${this._connectionParameters._endpoint}/report/${callContext.reportName}?_noRender=true&_context=${callContext.context}&_selection=${callContext.selection}&_selectionCount=${callContext.selectionCount}`,
+                headers: {
+                    'X-Security-Token': this._securityToken,
+                    'Cookie': '__sessiontoken=' + this._sessionToken, 
+                    'Content-Type': 'application/x-www-form-urlencoded'
+                },
+                method: 'POST',
+                credentials: 'include',
```

Comment:
Is this actually working. This was introduced by gaurav as a work around for the fact that cookies were not passerd automatically, and this was supposed to be removed quickly. I dont" like the fact that we're not solving the root cause of the issue and spreading this hack.

Reply:
This will be removed as soon as bearer token authentication is added in ACC. Team is currently working on it.